File: packages/web/src/lib/sentry.ts
/**
 * Sentry browser error tracking and performance monitoring.
 */
import * as Sentry from '@sentry/react'

export function initSentry(dsn: string, environment: string): void {
  Sentry.init({
    dsn,
    environment,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.replayIntegration(),
    ],
    tracesSampleRate: environment === 'development' ? 1.0 : 0.2,
    profilesSampleRate: 0.1,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  })
}

File: packages/api/app/integrations/sentry.py
"""Sentry error tracking and performance monitoring."""
import sentry_sdk
from sentry_sdk.integrations.fastapi import FastApiIntegration
from sentry_sdk.integrations.starlette import StarletteIntegration


def _before_send_transaction(event, hint):
    """Filter out health check transactions to reduce noise."""
    if event.get("transaction") in ("/health", "/healthz", "/ready", "/livez"):
        return None
    return event


def init_sentry(dsn: str, environment: str) -> None:
    """Initialise Sentry SDK with FastAPI integration.

    Args:
        dsn: Sentry project DSN.
        environment: Runtime environment name (development, staging, production).
    """
    traces_sample_rate = 1.0 if environment == "development" else 0.2

    sentry_sdk.init(
        dsn=dsn,
        environment=environment,
        integrations=[
            FastApiIntegration(),
            StarletteIntegration(),
        ],
        traces_sample_rate=traces_sample_rate,
        profiles_sample_rate=0.1,
        before_send_transaction=_before_send_transaction,
        send_default_pii=False,
    )
